File: fall-detector/src/main/java/altermarkive/guardian/Battery.java
package altermarkive.guardian;

import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.BatteryManager;

public class Battery {
    public static int level(Context context) {
        context = context.getApplicationContext();
        IntentFilter filter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
        Intent intent = context.registerReceiver(null, filter);
        int level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, 0);
        int scale = intent.getIntExtra(BatteryManager.EXTRA_SCALE, 1);
        level = (int) (level * 100.0 / scale);
        return (level);
    }
}

File: fall-detector/src/main/java/altermarkive/guardian/Battery.kt
package altermarkive.guardian

import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.BatteryManager

object Battery {
    internal fun level(context: Context): Int {
        val applicationContext: Context = context.applicationContext
        val filter = IntentFilter(Intent.ACTION_BATTERY_CHANGED)
        val intent = applicationContext.registerReceiver(null, filter) ?: return -1
        val level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, 0)
        val scale = intent.getIntExtra(BatteryManager.EXTRA_SCALE, 1)
        return (level * 100.0 / scale).toInt()
    }
}